File: web/src/api/normal/gost_client_forward.js
import request from "../../request/index.js";

const baseUrl = '/v1/normal/gost/client/forward'

export const apiNormalGostClientForwardList = () => {
    return request.request({
        url: `${baseUrl}/list`,
        method: 'POST',
    })
}

export const apiNormalGostClientForwardPage = (data) => {
    return request.request({
        url: `${baseUrl}/page`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientForwardCreate = (data) => {
    return request.request({
        url: `${baseUrl}/create`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientForwardRenew = (data) => {
    return request.request({
        url: `${baseUrl}/renew`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientForwardEnable = (data) => {
    return request.request({
        url: `${baseUrl}/enable`,
        method: 'POST',
        data
    })
}

export const apiNormalGostClientForwardUpdate = (data) => {
    return request.request({
        url: `${baseUrl}/update`,
        method: 'POST',
        data
    })
}


export const apiNormalGostClientForwardMatcher = (data) => {
    return request.request({
        url: `${baseUrl}/matcher`,
        method: 'POST',
        data
    })
}


export const apiNormalGostClientForwardAdmission = (data) => {
    return request.request({
        url: `${baseUrl}/admission`,
        method: 'POST',
        data
    })
}


export const apiNormalGostClientForwardDelete = (data) => {
    return request.request({
        url: `${baseUrl}/delete`,
        method: 'POST',
        data
    })
}

File: server/router/normal/gost_client_forward.go
package normal

import (
	"github.com/gin-gonic/gin"
	"server/controller/normal/gost_client_forward"
	"server/global"
	"server/router/middleware"
)

func InitGostClientForward(group *gin.RouterGroup) {
	g := group.Group("gost/client/forward", middleware.Auth(global.Jwt))
	g.POST("create", gost_client_forward.Create)
	g.POST("admission", gost_client_forward.Admission)
	g.POST("renew", gost_client_forward.Renew)
	g.POST("update", gost_client_forward.Update)
	g.POST("migrate", gost_client_forward.Migrate)
	g.POST("matcher", gost_client_forward.Matcher)
	g.POST("enable", gost_client_forward.Enable)
	g.POST("delete", gost_client_forward.Delete)
	g.POST("page", gost_client_forward.Page)
}
